### PR TITLE
docs: elucidate obscure discussion points

### PR DESCRIPTION
**Pull request by EpsilonPrime**
* Various description updates.
* Adds to Substrait Fiddle the list of third party tools.
* Changes discussion points to use admonitions to separate them better from the rest of the page.

### CODE REVIEW

File: site/mkdocs.yml [hunk 1]
```diff
@@ -94,7 +94,8 @@ markdown_extensions:
         - name: mermaid
           class: mermaid
           format: !!python/name:pymdownx.superfences.fence_code_format
-  - pymdownx.tabbed
+  - pymdownx.tabbed:
+      alternate_style: true
```
Comment (westonpace) [inline site/mkdocs.yml:98]: What does this do?

**Review by westonpace — CHANGES_REQUESTED** [review]
This looks pretty good.  Thanks for taking a stab at cleaning things up.

File: site/docs/relations/logical_relations.md [hunk 2]
```diff
@@ -448,6 +456,6 @@ The operator that defines modifications of a database schema (CREATE/DROP/ALTER
 %%% proto.algebra.DdlRel %%%
     ```
 
-## Discussion Points
+???+ "Discussion Points"
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/relations/logical_relations.md:459]: Yep, the question indicates the treatment of this section.

File: site/docs/relations/logical_relations.md [hunk 3]
```diff
@@ -393,7 +397,11 @@ The write operator is an operator that consumes one output and writes it to stor
 
 ### Write Definition Types
 
-Write definition types are built by the community and added to the specification. This is a portion of specification that is expected to grow rapidly.
+???+ info inline end "Adding new Write Definition Types"
+
+    If you have a read definition that's not covered here, see the [process for adding new write definition types](/governance/#substrait-voting-process).
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/relations/logical_relations.md:402]: Done.

File: site/docs/relations/logical_relations.md [hunk 4]
```diff
@@ -42,7 +42,11 @@ A filter expression must be interpreted against the direct schema before the pro
 
 ### Read Definition Types
 
-Read definition types are built by the community and added to the specification. This is a portion of specification that is expected to grow rapidly.
+???+ info inline end "Adding new Read Definition Types"
+
+    If you have a read definition that's not covered here, see the [process for adding new read definition types](/governance/#substrait-voting-process).
+
+Read definition types are built by the community and added to the specification.
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/relations/logical_relations.md:49]: Done.

File: site/docs/relations/basics.md [hunk 5]
```diff
@@ -70,9 +70,9 @@ A guarantee that data output from this operation is provided with a sort order.
...
 
-## Discussion Points
+???+ question "Discussion Points"
 
-* Do we try to make read definition types more extensible à la function signatures? Is that necessary if we have a custom relational operator?
-* How do we express decomposed types. For example, the Iceberg type above is for early logical planning. Once we do some operations, it may produce a list of Iceberg file reads. This is likely a secondary type of object.
-* We currently include a "generic properties" property on read type. Do we want this dumping ground?
+    * Should [read definition types](/relations/logical_relations/#read-definition-types) more extensible in the way that function signatures are? Are extensible read definition types necessary if we have custom relational operators?
+    * How are decomposed types expressed? For example, the Iceberg type above is for early logical planning. Once we do some operations, it may produce a list of Iceberg file reads. This is likely a secondary type of object.
+    * We currently include a "generic properties" property on read type. Do we want this dumping ground?
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/relations/basics.md:77]: Dropping for now.

File: site/docs/relations/basics.md [hunk 6]
```diff
@@ -70,9 +70,9 @@ A guarantee that data output from this operation is provided with a sort order.
...
 
 
-## Discussion Points
+???+ question "Discussion Points"
 
-* Do we try to make read definition types more extensible à la function signatures? Is that necessary if we have a custom relational operator?
-* How do we express decomposed types. For example, the Iceberg type above is for early logical planning. Once we do some operations, it may produce a list of Iceberg file reads. This is likely a secondary type of object.
-* We currently include a "generic properties" property on read type. Do we want this dumping ground?
+    * Should [read definition types](/relations/logical_relations/#read-definition-types) more extensible in the way that function signatures are? Are extensible read definition types necessary if we have custom relational operators?
+    * How are decomposed types expressed? For example, the Iceberg type above is for early logical planning. Once we do some operations, it may produce a list of Iceberg file reads. This is likely a secondary type of object.
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/relations/basics.md:76]: Not too clear to me either.  One thought I had was we could move this stuff into a commenting system (there is a mkdocs plugin that utilizes github discussions).  Long term these questions should be dealt with.

File: site/docs/relations/basics.md [hunk 7]
```diff
@@ -70,9 +70,9 @@ A guarantee that data output from this operation is provided with a sort order.
 
 
 
-## Discussion Points
+???+ question "Discussion Points"
 
-* Do we try to make read definition types more extensible à la function signatures? Is that necessary if we have a custom relational operator?
-* How do we express decomposed types. For example, the Iceberg type above is for early logical planning. Once we do some operations, it may produce a list of Iceberg file reads. This is likely a secondary type of object.
-* We currently include a "generic properties" property on read type. Do we want this dumping ground?
+    * Should [read definition types](/relations/logical_relations/#read-definition-types) more extensible in the way that function signatures are? Are extensible read definition types necessary if we have custom relational operators?
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/relations/basics.md:75]: Done.

File: site/docs/expressions/user_defined_functions.md [hunk 8]
```diff
@@ -1,3 +1,32 @@
...
+  -
+    name: "double"
+    description: "Double the value"
+    impls:
+      - args:
+          - name: x
+            value: fp32
+        options:
+          on_domain_error:
+            values: [ NAN, ERROR ]
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/expressions/user_defined_functions.md:23]: Done.

File: site/docs/expressions/subqueries.md [hunk 9]
```diff
@@ -65,3 +65,11 @@ WHERE x < ANY(SELECT y from t2)
 | Comparison operation  | The kind of comparison operation to use        | Yes      |
 | Expression            | Left-hand side expression to check             | Yes      |
 | Subquery              | Subquery to check                              | Yes      |
+
+
+
+=== "Protobuf Representation"
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/expressions/subqueries.md:71]: This syntax enables the superfences support where the contiguous sections merge together into a tabbed view (new behavior).

File: site/docs/expressions/embedded_functions.md [hunk 10]
```diff
@@ -19,10 +19,9 @@ The binary representation of an embedded function is:
     ```proto
 %%% proto.message.Expression.EmbeddedFunction %%%
     ```
+
 === "Human Readable Representation"
-    n/a
-=== "Example"
-    n/a
+    As the bytes are opaque to Substrait there is no equivalent human readable form.
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/expressions/embedded_functions.md:24]: Agreed.  I suspect we'll end up with a text format with the embedded function represented as a byte string in the long term.

File: site/docs/tools/third_party_tools.md [hunk 11]
```diff
@@ -3,4 +3,10 @@
 ## Substrait-tools
 The [substrait-tools](https://pypi.org/project/substrait-tools/) python package provides 
 a command line interface for producing/consuming substrait plans by leveraging the APIs
-from different producers and consumers.
\ No newline at end of file
+from different producers and consumers.# Substrait Fiddle
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/tools/third_party_tools.md:6]: Done.

File: site/mkdocs.yml (hunk 1 above)
Comment (EpsilonPrime) [reply site/mkdocs.yml:98]: This is an alternate implementation of the tabbed view that works better on narrow devices (like phones).

File: site/docs/types/type_parsing.md [hunk 12]
```diff
@@ -10,27 +10,57 @@ The components of this expression are:
...
+    list?<type>>
+    map<type0, type1>
+    ```
+
+=== "Text Format Examples"
+
+    ```
+    list?<list<string>>
+    list<struct<string, i32>>
+    map<i32?, list<map<i32, string?>>>
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/types/type_parsing.md:65]: I kept them to keep consistency with the other locations which have these sections even though the difference between the two here is slight.

File: site/docs/relations/logical_relations.md [hunk 13]
```diff
@@ -42,7 +42,11 @@ A filter expression must be interpreted against the direct schema before the pro
 
 ### Read Definition Types
 
-Read definition types are built by the community and added to the specification. This is a portion of specification that is expected to grow rapidly.
+???+ info inline end "Adding new Read Definition Types"
+
+    If you have a read definition that's not covered here, see the [process for adding new read definition types](/governance/#substrait-voting-process).
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/relations/logical_relations.md:47]: I'm adding a whole new section under the specification section called "Extending" where we can discuss these topics.  You may find some of the language there familiar.

File: site/docs/serialization/binary_serialization.md [hunk 14]
```diff
@@ -5,7 +5,7 @@ Substrait can be serialized into a [protobuf](https://developers.google.com/prot
 
 ## Plan
 
-The main top-level object used to communicate a Substrait plan using protobuf is a Plan message. The plan message is composed of a set of data structures that minimize repetition in the serialization along with one (or more) Relation trees. 
+The main top-level object used to communicate a Substrait plan using protobuf is a Plan message (see the [ExtendedExpression](/expressions/extended_expression/) for the other top-level object). The plan message is composed of a set of data structures that minimize repetition in the serialization along with one (or more) Relation trees.
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/serialization/binary_serialization.md:8]: Done.

File: site/docs/types/type_parsing.md [hunk 15]
```diff
@@ -10,27 +10,57 @@ The components of this expression are:
...
+    struct?[variation]<type0, type1,..., typeN>
+
+    # Named Struct
+    nstruct?[variation]<name0:type0, name1:type1,..., nameN:typeN>
+    ```
+
+=== "Text Format Examples"
+
+    ```
+    struct?<string, i8, i32?, timestamp_tz>
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/types/type_parsing.md:39]: Done.

File: site/docs/types/type_parsing.md [hunk 16]
```diff
@@ -10,27 +10,57 @@ The components of this expression are:
...
 | Parameters             | Compound types may have one or more configurable properties. The two main types of properties are integer and type properties. The parameters for each type correspond to a list of known properties associated with a type as declared in the order defined in the type specification. For compound types (types that contain types), the data type syntax will include nested type declarations. The one exception is structs, which are further outlined below. | Required where parameters are defined |
 
 ### Grammars
 
-It is relatively easy in most languages to produce simple parser & emitters for the type syntax. To make that easier, Substrait also includes an ANTLR [impl pending] grammar to ease consumption and production of types.
+It is relatively easy in most languages to produce simple parser & emitters for the type syntax. To make that easier, Substrait also includes an [ANTLR grammar](https://github.com/substrait-io/substrait-cpp/blob/main/src/substrait/textplan/parser/grammar/SubstraitPlanParser.g4#L108) to ease consumption and production of types.
```
[the remark this answers is not in the document]
Comment (EpsilonPrime) [reply site/docs/types/type_parsing.md:20]: Added more.

**Comment by EpsilonPrime** [comment]
@westonpace Any further thoughts here?

**Review by jacques-n — APPROVED** [review]

**Pull request merged**